internal/backends/python/python.go

```diff
@@ -825,6 +841,17 @@ func makePythonUvBackend() api.LanguageBackend {
 
 				cmd = append(cmd, pep440Join(name, spec))
 			}
+
+			for _, index := range extraIndexes {
+				if strings.HasPrefix(runtime.GOOS, index.os) {
+					cmd = append(cmd, "--index", index.url)
```

Comment:
Were there issues using the envvar approach? This seems to add an additional index to search for every package install, instead of scoping it down to just when trying to install pytorch

Reply:
Not sure if I'm doing it wrong but I think doing `UV_INDEX=pytorch=https://download.pytorch.org/whl/cpu uv add torch` also does the same thing as `--index` where it adds the index to the `pyproject.toml`. I think `--extra-index-url` doesn't but it shows a warning to use `index` instead

Reply:
Ah! Looking more closely at the docs, I completely agree with you.

I do think it would be cleaner for the user if we were to use the envvar approach, as that won't show up in the packager pane (consider that `uv add pytest` would end up with `--index .../pytorch/...` in the command that we echo back in the console). No doubt we'd end up with support tickets asking why pytorch is involved at all.

---

The one additional observation I will make is that no matter which strategy we use to supply the extra indices, if that index is _down_ then uv fails to resolve at all.

We should conditionally supply these values based on whether the package is currently being installed or is in the existing dependencies.

We can exclude transitive dependencies for now and share with support that by promoting pytorch to be a direct dependency may resolve issues for certain libraries. If this is a big issue we can brainstorm other strategies for conditionally enabling that index, or man-in-the-middle uv failures and switch strategies based on whether an attempted invocation is successful.

Reply:
Sounds good! Updated to use the env var (prepending to it if necessary) and made a mapping from package names to extra indexes to use so that we only override on installing those packages or we have those packages when doing `uv sync`.

Also removed `--index-strategy unsafe-best-match` for now but let me know if you think we should keep it.

Reply:
Definitely need to put the `UV_INDEX_STRATEGY=unsafe-best-match` in there, otherwise we aren't guaranteed to get a functional version of pytorch (at least given my testing)